sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobSasConstructor.java

```diff
@@ -0,0 +1,503 @@
+         themselves, we cheat except for the signature value.
+         */
+        StringBuilder sb = new StringBuilder();
+
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SERVICE_VERSION, this.version);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_PROTOCOL, this.protocol);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_START_TIME, formatQueryDate(this.startTime));
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_EXPIRY_TIME, formatQueryDate(this.expiryTime));
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_IP_RANGE, this.ipRange);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_IDENTIFIER, this.identifier);
+
+        if (key != null) {
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_OBJECT_ID, key.getSignedOid());
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_TENANT_ID, key.getSignedTid());
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_KEY_START,
+                formatQueryDate(key.getSignedStart()));
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_KEY_EXPIRY,
+                formatQueryDate(key.getSignedExpiry()));
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_KEY_SERVICE, key.getSignedService());
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_KEY_VERSION, key.getSignedVersion());
+        }
+
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_RESOURCE, resource);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_PERMISSIONS, permissions);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNATURE, signature);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_CACHE_CONTROL, this.cacheControl);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_CONTENT_DISPOSITION, this.contentDisposition);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_CONTENT_ENCODING, this.contentEncoding);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_CONTENT_LANGUAGE, this.contentLanguage);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_CONTENT_TYPE, this.contentType);
+
+        return sb.toString();
+    }
+
+    private void tryAppendQueryParameter(StringBuilder sb, String param, Object value) {
+        if (value != null) {
+            if (sb.length() != 0) {
+                sb.append('&');
+            }
+            sb.append(Utility.urlEncode(param)).append('=').append(Utility.urlEncode(value.toString()));
+        }
+    }
+
+    private String formatQueryDate(OffsetDateTime dateTime) {
+        if (dateTime == null) {
+            return null;
+        } else {
+            return Utility.ISO_8601_UTC_DATE_FORMATTER.format(dateTime);
+        }
```

Comment:
We ain't paying for all these lines of code! Use ternary operator.

Reply:
Will cleanup if the class is kept